ra/ra.go

```diff
@@ -2167,6 +2169,9 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 		if err != nil {
 			return nil, err
 		}
+		if authzIDs.Ids == nil {
+			return nil, errIncompleteGRPCResponse
+		}
```

Comment:
> And it may be worth throwing a more specific and descriptive error.

I understand where you're coming from but I don't believe that accurately reflects what would have happened before the unwrapping. This would be introducing new behavior.

> I think we're better off checking if `len(authzIDs.Ids) == 0`.

Checking `len()` on slices that could have been initialized is good as there's a chance that they would be empty but not be `nil`. But wouldn't an empty slice via gRPC always be `nil`? If this is the case, is `nil` not equal to `len(<field>) == 0`?

Reply:
@aarongable and I chatted about this in a call and arrived at:

> I understand where you're coming from but I don't believe that accurately reflects what would have happened before the unwrapping. This would be introducing new behavior.

Yes, but since we're not depending on `errIncompleteResponse` as a sentinel this wouldn't be that significant a change in behavior.

> I think we're better off checking if len(authzIDs.Ids) == 0.

1. Without confirming that protobuff contract guarantees an empty array will be transmitted as the zero value the safest course is to check the `len()`
2. Checking `len()` is, in general, more readable